### ISSUE TITLE
Cannot enable tunnel service when at least 1 client is not configured

### ISSUE THREAD

**Issue by rwayboy** [micahmo/WireGuardServerForWindows#26]
Hello. I would like to make a suggestion about the application.

Would it be possible for us to still be able to enable the tunnel service even when both client folders (clients_data and clients_wg) are empty? The reason I am asking this is because, after I generate the config files, I don't want to keep them on the server machine. I only want each client to keep his config file (for security purposes).

Thank you.

**Comment by micahmo** [comment]
Hi @rwayboy, good suggestion, that makes total sense. I will see if I can add it.

**Comment by micahmo** [comment]
Hi, I've just pushed a new version ([v1.7.1](https://github.com/micahmo/WireGuardServerForWindows/releases/tag/v1.7.1)) that should address this issue.

I removed the restriction that prevented the tunnel from being installed without clients configured. As advertised, the application was already supposed to update the tunnel with any configuration changes, so there was no reason for that restriction. (See [here](https://github.com/micahmo/WireGuardServerForWindows#tunnnel-service).)

That said, I discovered that this actually didn't work as expected, due to a change for #20, so I've rolled that back.

Those two changes should address your specific request. However, your use case gave me another idea, which is to allow you to save clients without the private key. It will no longer be a required field, so your workflow could be:
1. Create client
2. Export client config (QR code or file)
3. Delete private key and save

This ensures that:
* No private client info is saved on the server
* The client configurations are considered valid by WS4W
* The server config still has all the info it needs to accept client connections

In summary, the changes are:
1. Allow tunnel to be installed with no clients configured
2. Fix issue where client changes were not propagated to running tunnel
3. Allow clients to be saved without private key

Can you try the latest version and let me know if this helps?

**Comment by rwayboy** [comment]
Awesome, I wasn't expecting an update right away haha. Thank you very much, the tunnel service can now be enabled without client configs.

By the way, I have two more suggestions in case you are interested, can I post them here or create a new thread instead?

**Comment by micahmo** [comment]
No worries! I will close this issue as it is now resolved.

Feel free to create a new thread (or 2, if the suggestions aren't related).

Fair warning that I won't always be able to update that quickly, especially if the suggestions are more elaborate. 😊
